### PR TITLE
Feature/upload image files

### PR DESCRIPTION
**Pull request by samshara**
Add support to add image upload for `event_map` and `images` field in DREF Application

### CODE REVIEW

File: src/root/views/DrefApplicationForm/index.tsx [hunk 1]
```diff
@@ -232,9 +232,9 @@ function DrefApplication(props: Props) {
   const {
     pending: drefApplicationPending,
   } = useRequest<DrefApiFields>({
-    skip: !drefId,
-    url: `api/v2/dref/${drefId}/`,
-    onSuccess: (response) => {
+      skip: !drefId,
+      url: `api/v2/dref/${drefId}/`,
+      onSuccess: (response) => {
```
Comment (tnagorra) [inline src/root/views/DrefApplicationForm/index.tsx:237]: These whitespace changes should be reverted

File: src/root/views/DrefApplicationForm/index.tsx [hunk 2]
```diff
@@ -312,12 +312,11 @@ function DrefApplication(props: Props) {
 
     if (currentStep === 'submission') {
       if (finalValues && userDetails && userDetails.id) {
-        const body = {
+        const  body = {
```
Comment (tnagorra) [inline src/root/views/DrefApplicationForm/index.tsx:315]: This whitespace change should be reverted

File: src/root/views/DrefApplicationForm/useDrefFormOptions.ts [hunk 3]
```diff
@@ -88,6 +88,11 @@ export function validEmailCondition(email: any) {
   return isDefined(email) && !emailRegex.test(email)
     ? 'Invalid Email' : undefined;
 }
+export function lessThanSixItemsCondition(value: any) {
```
Comment (tnagorra) [inline src/root/views/DrefApplicationForm/useDrefFormOptions.ts:91]:
```suggestion
export function lessThanSixImagesCondition(value: any) {
```

File: src/root/lang.js [hunk 4]
```diff
@@ -1617,7 +1617,9 @@ export default {
   timeFrames: 'Timeframes',
   eventDate: 'Date of event',
   goFieldReportDate: 'Go Field Report Date',
+  triggerDate: 'Disaster date/ trigger date',
```
Comment (tnagorra) [inline src/root/lang.js:1620]:
```suggestion
  triggerDate: 'Disaster date / trigger date',
```

File: src/root/views/DrefApplicationForm/EventDetails/index.tsx [hunk 5]
```diff
@@ -250,6 +251,21 @@ function EventDetails(props: Props) {
...
+            value={value.images}
+            onChange={onValueChange}
+            accept="image/*"
+            multiple
+            showStatus
+            error={error?.fields?.images}
+          >
+            Upload
+          </GoFileInput>
+        </InputSection>
```
Comment (tnagorra) [inline src/root/views/DrefApplicationForm/EventDetails/index.tsx:268]: Don't we need to show the uploaded file during edit?

File: src/root/utils/restRequest/go.ts [hunk 6]
```diff
@@ -96,20 +121,36 @@ export const processGoOptions: GoContextInterface['transformOptions'] = (
...
+  const requestBody = body ? JSON.stringify(body) : undefined;
+  return {
+    method,
+    headers: {
+      Accept: isCsvRequest ? CONTENT_TYPE_CSV : CONTENT_TYPE_JSON,
+      'Content-Type': isCsvRequest
+        ? 'text/csv; charset=utf-8'
+        : 'application/json; charset=utf-8',
+      Authorization: token ? `Token ${token}` : '',
+      'Accept-Language': (enforceEnglish || method !== 'GET') ? 'en' : currentLanguage,
```
Comment (tnagorra) [inline src/root/utils/restRequest/go.ts:148]: We should reuse these variable calcuation

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/root/views/DrefApplicationForm/EventDetails/index.tsx (hunk 5 above)
Comment (samshara) [reply src/root/views/DrefApplicationForm/EventDetails/index.tsx:268]: As discussed, let's create an issue for this later and work on it.

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**